pkg/operator/status.go

```diff
@@ -48,6 +48,12 @@ func isDeploymentStatusAvailable(deploy *appsapi.Deployment) bool {
 	return deploy.Status.AvailableReplicas > 0
 }
 
+func isDeploymentStatusAvailableAndUpdated(deploy *appsapi.Deployment) bool {
+	return deploy.Status.AvailableReplicas > 0 &&
+		deploy.Status.ObservedGeneration >= deploy.Generation &&
+		deploy.Status.UpdatedReplicas == deploy.Status.Replicas
```

Comment:
I think it's still possible that you see the old deployment version here, since we use lister caches.  (or for that matter if we just whiff completely on updating the deployment to the new image version).

I'm going to pull this commit into a new PR and iterate on it.  We need to actually confirm that the deployment which is available also reflects the new version.